crates/nu-command/src/filesystem/ls.rs

```diff
@@ -219,6 +225,69 @@ impl Command for Ls {
     }
 }
 
+impl Ls {
+    fn apply_post_sorting(
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        result: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let sort_by_configs = stack.get_config(engine_state).ls.sort_by.clone();
+        let cloned_metadata = result.metadata().clone();
+        let mut comparator_vals: Vec<Value> = result.into_iter().collect();
+
+        for sort_by_config in sort_by_configs {
+            let compare_err =
+                Self::setup_and_sort_by_config(call, &mut comparator_vals, &sort_by_config);
+
+            if let Some(compare_err_unwrapped) = compare_err {
+                return Err(compare_err_unwrapped);
+            }
+        }
+
+        let val = Value::list(comparator_vals, call.head);
+        let data = val.into_pipeline_data_with_metadata(cloned_metadata);
+
+        Ok(data)
+    }
+
+    fn setup_and_sort_by_config(
+        call: &Call,
+        comparator_vals: &mut [Value],
+        sort_by_config: &str,
+    ) -> Option<ShellError> {
+        let words_in_sort_by_config: Vec<_> = sort_by_config.split_whitespace().collect();
+        let mut comparators = [Comparator::CellPath(CellPath {
+            members: vec![PathMember::string(
+                words_in_sort_by_config[0].to_string(),
+                false,
+                Span::unknown(),
+            )],
+        })];
+        let contains_insensitive = sort_by_config.contains("ignore-case"); // TODO expand
+        let contains_naturel = sort_by_config.contains("natural"); // TODO expand
+        let contains_reverse = sort_by_config.contains("reverse"); // TODO expand
+
+        let mut compare_err: Option<ShellError> = None;
+        comparator_vals.sort_by(|a, b| {
+            compare_by(
+                a,
+                b,
+                &mut comparators,
+                call.head,
+                contains_insensitive,
+                contains_naturel,
+                &mut compare_err,
+            )
+        });
```

Comment:
Could you use `sort_utils::sort_by` here instead? It does the same thing for you, less code duplication 😄